### ISSUE TITLE
Downloadonly does not save files to disk if repo url starts with file://

### ISSUE THREAD

**Issue by Arash-Afshar** [rpm-software-management/yum#83]
I traced the code path and it does not seem to be a bug, rather a feature. I think it makes sense to have this as a default behavior since the files are presumably already on the disk so there is no point in saving them again.

But, in my use case, which I will describe shortly, it makes more sense to actually save them on the disk. Therefore, it would be great to be able to override this default behavior either with a plugin or an input flag. In my use case I am serving very large packages over an io device that is mounted to a directory. Since the rpms are large and io device is very slow I want to first issue a `yum --downloadonly` overnight to fetch the rpms and then install them in the morning. What ends up happening is that calling yum with downloadonly will actually read and download the large rpms but does not save them locally. Then, in the morning yum will download everything again and installs them. 

I have tracked the required code change to fix this to the following two lines:

1. If "copy_local=1" flag is passed to https://github.com/rpm-software-management/yum/blob/1222f377b8dcecb77456ea378a28e1dc23ba4207/yum/yumRepo.py#L1065 it will download the files.
2. The earliest time to set this flag would be in https://github.com/rpm-software-management/yum/blob/1222f377b8dcecb77456ea378a28e1dc23ba4207/yum/__init__.py#L2591 as a key-value arguement to be passed to the getPackage function

This could change would require me to fork yum, which I rather not do. So, my preference is
1. Submit a PR to make this change and have a flag to control whether to always download or not
2. Write a yum plugin that takes care of this. I have looked at the yum plugins but they do not seem to be capable of handling such cases. If this can be done with a plugin, please let me know.

I would appreciate any feedback on this. Thank you!

------
EDIT:

btw, I have seen the local.py plugin: http://yum.baseurl.org/gitweb?p=yum-utils.git;a=tree;f=plugins/local;h=27633ae6037ea85aa83c9f939397632f9e74a6cb;hb=HEAD

It has two limitations:
1. It does not work with the --downloadonly flag. Since when downloadonly is passed, posdownload hook is not called. This can be fixed by modifiying local.py plugin to mimic the behaviour of downloadonly and stop using downloadoly.
2. It would still download the file twice. First, when it normally downloads it and second in the postdownload hook where it copies the file to local dir. Therefore, it does not solve the problem of downloading a big file over a slow device twice, but at least pushes these downloads over night.

So basically I am looking for a better solution than the above.

**Comment by dmnks** [comment]
Thank you for the report. However, please note that this project has been deprecated and is no longer maintained upstream. Please consider migrating to [DNF](https://github.com/rpm-software-management/dnf) which is the successor of YUM. That said, I’ll close this issue now.
